### PR TITLE
Issue#295 delivery send tx

### PR DESCRIPTION
**Pull request by pldespaigne**
- [x] sending a simple transaction from the relayer to the user's contract with the hash of the delivry id in the data field
- [x] update the stakeholder firestore with the tx hash
- [x] display a link to etherscan on the signed delivery component

### CODE REVIEW

File: functions/src/relayer.ts [hunk 1]
```diff
@@ -176,4 +176,25 @@ export const relayerRequestTokensLogic = async (
   const tx = await relayer.wallet.sendTransaction({to: fullName, value: weiAmount});
   console.log(`tx sent (request tokens) : ${tx.hash}`); // display tx to firebase logging
   return tx;
-};
\ No newline at end of file
+};
+
+
+export const relayerSignDeliveryLogic = async (
+  { username, deliveryId, stakeholderId }: { username: string; deliveryId: string, stakeholderId: string },
```
Comment (GrandSchtroumpf) [inline functions/src/relayer.ts]: Create a type for that.

File: functions/src/relayer.ts [hunk 2]
```diff
@@ -176,4 +176,25 @@ export const relayerRequestTokensLogic = async (
...
+  config: any
```
Comment (GrandSchtroumpf) [inline functions/src/relayer.ts]: Don't use any. Create a type if needed.

File: functions/src/relayer.ts [hunk 3]
```diff
@@ -176,4 +176,25 @@ export const relayerRequestTokensLogic = async (
...
+  if (!username || !deliveryId || !stakeholderId) {
+    throw new Error('"username", "deliveryId" and "stakeholderId" are mandatory parameters !');
+  }
+  const relayer: Relayer = initRelayer(config);
+  // compute needed values
+  const fullName = `${username}.${config.relayer.basedomain}`;
+  const hash = utils.keccak256(utils.toUtf8Bytes(deliveryId));
+  const tx = await relayer.wallet.sendTransaction({to: fullName, data: hash});
+
+  await db.collection('deliveries').doc(deliveryId).collection('stakeholders').doc(stakeholderId).update({tx: tx.hash});
```
Comment (GrandSchtroumpf) [inline functions/src/relayer.ts]: use 
```typescript
db.doc(`deliveries/${deliveryId}/stakholders/${stakeholderId}`).update({tx: tx.hash})
```

File: libs/auth/src/lib/+state/auth.service.ts [hunk 4]
```diff
@@ -114,6 +114,14 @@ export class AuthService {
     }
   }
 
+  public async signDelivery(deliveryId: string, stakeholderId: string) {
+    try {
+      return this.wallet.signDelivery(deliveryId, stakeholderId);
+    } catch (error) {
+      console.error('The relayer has failed to forward the signature to the blockchain');
```
Comment (GrandSchtroumpf) [inline libs/auth/src/lib/+state/auth.service.ts]: This error should be in the `signDelivery` of the relayer.
Why don't you inject direcly the wallet in the component ?

File: functions/src/relayer.ts (hunk 1 above)
Comment (pldespaigne) [reply functions/src/relayer.ts]: ✔️

File: functions/src/relayer.ts (hunk 2 above)
Comment (pldespaigne) [reply functions/src/relayer.ts]: ✔️

File: functions/src/relayer.ts (hunk 3 above)
Comment (pldespaigne) [reply functions/src/relayer.ts]: ✔️

File: libs/auth/src/lib/+state/auth.service.ts (hunk 4 above)
Comment (pldespaigne) [reply libs/auth/src/lib/+state/auth.service.ts]: ✔️

**Pull request merged**